server/svix-server/src/queue/mod.rs

```diff
@@ -40,11 +40,67 @@ impl MessageTask {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MessageTaskBatch {
+    pub msg_id: MessageId,
+    pub app_id: ApplicationId,
+    pub endpoint_ids: Vec<EndpointId>,
+    pub trigger_type: MessageAttemptTriggerType,
+    pub attempt_count: u16,
+}
+
+impl MessageTaskBatch {
+    pub fn new_task(
+        msg_id: MessageId,
+        app_id: ApplicationId,
+        endpoint_ids: Vec<EndpointId>,
+        trigger_type: MessageAttemptTriggerType,
+    ) -> QueueTask {
+        QueueTask::MessageBatchV1(Self {
+            msg_id,
+            app_id,
+            endpoint_ids,
+            attempt_count: 0,
+            trigger_type,
+        })
+    }
+    pub fn to_message_task(self, endpoint_id: &EndpointId) -> MessageTask {
+        MessageTask {
+            msg_id: self.msg_id,
+            app_id: self.app_id,
+            endpoint_id: endpoint_id.clone(),
+            trigger_type: self.trigger_type,
+            attempt_count: self.attempt_count,
+        }
+    }
+}
+
 #[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 #[serde(tag = "type")]
 pub enum QueueTask {
     MessageV1(MessageTask),
+    MessageBatchV1(MessageTaskBatch),
+}
+
+impl QueueTask {
+    pub fn get_message_id(self) -> MessageId {
```

Comment:
```suggestion
    pub fn msg_id(self) -> MessageId {
```
Rust doesn't uses a `get_` prefix for getters (convention). Also, we call it msg_id everywhere.